src/lib.rs

```diff
@@ -2,6 +2,17 @@ use pgrx::prelude::*;
 
 pgrx::pg_module_magic!();
 
+/// inspired from https://docs.rs/pgrx-pg-sys/0.11.4/src/pgrx_pg_sys/submodules/elog.rs.html#243-256
+macro_rules! error_code {
```

Comment:
just a question to better understand the `ereport` implications -- is it expected that:
> This will abort the current Postgres transaction.

(from https://docs.rs/pgrx-pg-sys/0.11.4/src/pgrx_pg_sys/submodules/elog.rs.html#243)

so to my understanding if I do `BEGIN` and later e.g. `verify_signature()` fails then `ROLLBACK` will be performed - is my assumption correct?

maybe I'm missing context on postgres internals but my assumption is based on https://www.postgresql.org/docs/current/sql-abort.html
> ABORT — abort the current transaction
> (...)
> This command is identical in behavior to the standard SQL command [ROLLBACK](https://www.postgresql.org/docs/current/sql-rollback.html), and is present only for historical reasons.

Reply:
... I think "abort the current Postgres transaction" originates from https://docs.rs/pgrx/latest/pgrx/macro.ereport.html
> if it’s a [PgLogLevel::ERROR](https://docs.rs/pgrx/latest/pgrx/enum.PgLogLevel.html#variant.ERROR) or greater, Postgres’ error handling takes over and, in the case of [PgLogLevel::ERROR](https://docs.rs/pgrx/latest/pgrx/enum.PgLogLevel.html#variant.ERROR), aborts the current transaction.

which I understand in a different way compared to https://www.postgresql.org/docs/current/error-message-reporting.html
> If the severity level is ERROR or higher, ereport aborts execution of the current query and does not return to the caller.

... so self answering my prev question -- seams like nothing to worry about here

Reply:
This is correct. Although we only expect the `auth.user_id()` or `auth.session(key)` function to be run by users in a transaction, and currently only the user_id fn can fail if the sub field is not a string. The signature verification occurs before any transactions automatically by local_proxy in our system